### PR TITLE
Don't 404 on accounts GET

### PR DESCRIPTION
**Pull request by kddejong**
## Proposed changes

- No more 404s on lease GETs

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (changes to code, which do not change application behavior)

## Checklist

- [ ] I have filled out this PR template
- [ ] I have read the [CONTRIBUTING](https://github.com/Optum/dce/blob/master/docs/CONTRIBUTING.md) doc
- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (`README.md`, inline comments, etc.)
- [ ] I have updated the `CHANGELOG.md` under a `## next` release, with a short summary of my changes

### CODE REVIEW

File: tests/acceptance/api_test.go [hunk 1]
```diff
@@ -1559,6 +1559,17 @@ func TestApi(t *testing.T) {
...
 
+		t.Run("When there is a principal ID and an Account ID parameter", func(t *testing.T) {
+			resp := apiRequest(t, &apiRequestInput{
+				method: "GET",
+				url:    apiURL + "/leases?principalId=" + principalIDOne + "&accountId=" + accountIDOne,
+				json:   nil,
+			})
+
+			results := parseResponseArrayJSON(t, resp)
+			assert.Equal(t, 1, len(results), "only one lease should be returned")
```
Comment (eschwartz) [inline tests/acceptance/api_test.go]: do we have test coverage for empty results? Might be a good idea...

File: cmd/lambda/leases/get.go [hunk 2]
```diff
@@ -49,14 +49,14 @@ func GetLeasesByPrincipcalIDAndAccountID(w http.ResponseWriter, r *http.Request)
 		response.WriteServerErrorWithResponse(w, errMsg)
 		return
 	}
-	if lease == nil {
-		log.Printf("Error Getting Lease for Id: %s", err)
-		response.WriteNotFoundError(w)
-		return
+	leaseResponses := []*response.LeaseResponse{}
```
Comment (nathanagood) [inline cmd/lambda/leases/get.go]: So one object was coming back but now an array is coming back. Does that break the contract?

File: tests/acceptance/api_test.go (hunk 1 above)
Comment (nathanagood) [reply tests/acceptance/api_test.go]: It was a single object but is now an array of one. Is this really a valid test that asserts same-as-before-the-break functionality?

File: cmd/lambda/leases/get.go (hunk 2 above)
Comment (eschwartz) [reply cmd/lambda/leases/get.go]: Yes. Though I should point out that at one point we _were_ returning empty lists instead of a 404, (as of v0.23.0, at least). So we more recently broke the contract, this change is putting it back.

Comment (eschwartz) [reply cmd/lambda/leases/get.go]: sorry, I got confused here. The API that broke at some point was the `GET /leases?principalId=`
we'd always had `GET /leases?principalId=&accountId=` return a single object (or 404),
as legacy behavior from before we had Lease IDs.

But I would like to change that, and I'm ok calling that a breaking change to our API.

File: tests/acceptance/api_test.go (hunk 1 above)
Comment (AmudaPalani) [reply tests/acceptance/api_test.go]: Can we add a test to verify an empty array is returned when no record found?

File: tests/acceptance/api_test.go [hunk 3]
```diff
@@ -1559,6 +1559,28 @@ func TestApi(t *testing.T) {
...
+		})
+
+		t.Run("When there is no leases found with principal and account don't exist", func(t *testing.T) {
+			resp := apiRequest(t, &apiRequestInput{
+				method: "GET",
+				url:    apiURL + "/leases?principalId=reallybadprincipal&accountId=notanaccount",
+				json:   nil,
+			})
+
+			results := parseResponseArrayJSON(t, resp)
```
Comment (eschwartz) [inline tests/acceptance/api_test.go]: 👍
